jobsdb/jobsdb.go

```diff
@@ -1890,6 +1893,30 @@ func (jd *Handle) GetToProcess(ctx context.Context, params GetQueryParams, more
 	return &MoreJobsResult{JobsResult: JobsResult{Jobs: list, LimitsReached: unprocessed.LimitsReached}, More: mtoken}, nil
 }
 
+func (jd *Handle) GetToProcess(ctx context.Context, params GetQueryParams, more MoreToken) (*MoreJobsResult, error) { // skipcq: CRT-P0003
+
+	if !jd.config.GetBool("JobsDB.useSingleGetJobsQuery", true) { // TODO: remove condition after successful rollout of sinle query
+		return jd.getToProcessLegacy(ctx, params, more)
+	}
+
+	if params.JobsLimit == 0 {
+		return &MoreJobsResult{More: more}, nil
+	}
+	params.stateFilters = []string{Failed.State, Waiting.State, Unprocessed.State}
+	slices.Sort(params.stateFilters)
```

Comment:
Aren't we creating the slice just one line above? Can't we create it in a sorted manner already? Also, do we need it to be sorted?

Reply:
Instead of a comment (e.g. this slice needs to be lexicographically sorted), I prefer to use the `slices.Sort` operation explicitly. 
If this slice needs to be changed in the future, relying on a person to make sure the new slice will be sorted is riskier than using `slices.Sort`